azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_nuget.go

```diff
@@ -0,0 +1,194 @@
+				Required:         true,
+				Sensitive:        true,
+				DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+			},
+			patHashKey: patHashSchema,
+		},
+	}
+
+	keyHashKey, keyHashSchema := tfhelper.GenerateSecreteMemoSchema("key")
+	ak := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Description:      "The Nuget Feed API key.",
+				Type:             schema.TypeString,
+				Required:         true,
+				Sensitive:        true,
+				DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+			},
+			keyHashKey: keyHashSchema,
+		},
+	}
+
+	patHashKeyU, patHashSchemaU := tfhelper.GenerateSecreteMemoSchema("username")
+	patHashKeyP, patHashSchemaP := tfhelper.GenerateSecreteMemoSchema("password")
+	aup := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"username": {
+				Description:      "The Nuget feed user name.",
+				Type:             schema.TypeString,
+				Required:         true,
+				Sensitive:        true,
+				DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+			},
+			patHashKeyU: patHashSchemaU,
+			"password": {
+				Description:      "The Nuget feed password.",
+				Type:             schema.TypeString,
+				Required:         true,
+				Sensitive:        true,
+				DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+			},
+			patHashKeyP: patHashSchemaP,
+		},
+	}
+
+	r.Schema["authentication_token"] = &schema.Schema{
+		Type:         schema.TypeList,
+		Optional:     true,
+		MinItems:     1,
+		MaxItems:     1,
+		Elem:         at,
+		ExactlyOneOf: []string{"authentication_basic", "authentication_token", "authentication_none"},
+	}
+
+	r.Schema["authentication_none"] = &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MinItems: 1,
+		MaxItems: 1,
+		Elem:     ak,
+	}
+
+	r.Schema["authentication_basic"] = &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MinItems: 1,
+		MaxItems: 1,
+		Elem:     aup,
+	}
+
+	return r
+}
+
+// Convert internal Terraform data structure to an AzDO data structure
+func expandServiceEndpointNuget(d *schema.ResourceData) (*serviceendpoint.ServiceEndpoint, *uuid.UUID, error) {
+	serviceEndpoint, projectID := doBaseExpansion(d)
+	serviceEndpoint.Type = converter.String("externalnugetfeed")
+	serviceEndpoint.Url = converter.String(d.Get("url").(string))
+	authScheme := "Token"
+
+	authParams := make(map[string]string)
+
+	if x, ok := d.GetOk("authentication_token"); ok {
+		authScheme = "Token"
+		msi := x.([]interface{})[0].(map[string]interface{})
+		authParams["apitoken"] = expandSecret(msi, "token")
+	} else if x, ok := d.GetOk("authentication_none"); ok {
+		authScheme = "None"
+		msi := x.([]interface{})[0].(map[string]interface{})
+		authParams["nugetkey"] = expandSecret(msi, "key")
+	} else if x, ok := d.GetOk("authentication_basic"); ok {
+		authScheme = "UsernamePassword"
+		msi := x.([]interface{})[0].(map[string]interface{})
+		authParams["username"] = expandSecret(msi, "username")
```

Comment:
Don't use expandSecret
```suggestion
		authParams["username"], ok = msi["username"].(string)
		if !ok {
			return nil, nil, errors.New("Unable to read 'username'")
		}
```